structure/redblacktree/rbtree_test.go

```diff
@@ -0,0 +1,107 @@
+package redblacktree
```

Comment:
Could I ask to make this one `redblacktree_test` instead. I want to see the implementation where only exposed api is tested because currently I think even your RBNode would not be working as expected when you make this change. Try it.

Reply:
I can't even think of how a RBTree will be made outside of this package.

Reply:
The only issue is that since more methods/functions are exposed, I will have to write tests for them, will add them in an upcoming commit